### ISSUE TITLE
Voiceover screen reader - not consistently able to open links

### ISSUE THREAD

**Issue by ruettet** [signalapp/Signal-iOS#5968]
(on behalf of https://disabled.social/@JustGrist/113986968822757833, with my fullest support)

“Hello,

Firstly, I want to thank the Signal devs for making sure that the IOS app is really very accessible for blind IOS users who use the Voiceover screen reader. That said, I noticed that while using Voiceover, I am not consistently able to open links sent to me in messages. Double tapping on the message doesn’t work. …

There is also no option for opening links in the options Voiceover users get when swiping up and down on a message or any option to open links in the context menu that pops up when I double tap and hold on a message. Adding an option for opening links either to the options that Voiceover users get when swiping up and down on a message or to the pop-up context menu would be the best way to ensure that blind IOS users have full access to the Signal app’s functionality.

Thank you.”

**Comment by PeterSaathoff-Harshfield** [comment]
I also experience this issue. If the URL is the only text in the message, VoiceOver can activate it. If there's any plain text in addition to a URL, the link can't be activated.

**Comment by noatamir** [comment]
Please address this issue

**Comment by atSquiDevONtwitter** [comment]
Hello,

I've just tested your issue in the latest build of Signal for iOS.

I've noticed that using VoiceOver, opening links sent to me in messages is inconsistent. Double tapping on the message doesn’t always work, and there’s no option to open links when swiping up and down on a message. There's also no link-opening option in the context menu that appears when I double tap and hold on a message.

Other VoiceOver users have also reported issues like VoiceOver becoming muted or slow, the cursor jumping to the top of the page, problems with the app switcher, and general unresponsiveness. These issues can be frustrating, and many users have noticed them across different iOS versions. Apple is aware of these problems, and users are encouraged to report them through the Feedback Assistant tool.

[TO: Signal Devs]
I believe addressing these accessibility issues would significantly enhance the user experience for VoiceOver users. Thank you for your ongoing efforts to improve accessibility.

**Comment by PeterSaathoff-Harshfield** [comment]
This bug is still valid. It should not be closed.

Because of this bug, I still have to tell people that if they want to send me a link, they must put it in its own message, rather than along with other text.

**Comment by aubrielee** [comment]
It would be great if the Signal team could address this, of course primarily for the people who use VoiceOver, and secondarily for the people like me who want to text those people using Signal without having to change how we write text messages just for the one app (Signal) that has strange link behavior.

**Comment by kinoroy** [comment]
I was able to reproduce this, but it doesn't seem to be caused by whether or not there is additional text within the message in addition to the URL. I was able to activate the link by double-tapping regardless of additional text in the message. However I can reproduce this in the case where there is no link preview generated for the link; double tapping in this case does nothing. 

Since the accessibility label for messages doesn't say whether the message contains a link preview or not, if the user cannot see the message they would have no way of knowing if there is a link preview (which is decided by the sending user). I can see how that would be confusing as two messages that sound the exact same to a VoiceOver user, have different accessibility actions, one opens a link, and one does nothing. 

Attached screen recording showing the link can be activated in both cases where a link preview is generated, but cannot be activated when there is no link preview for the message:

https://github.com/user-attachments/assets/e5451ac5-5a7b-47b9-a07b-8287a6e46424

[Video Description: A screen recording of an iPhone using Signal with VoiceOver enabled, the user selects a message containing only a link to Signal.org as text, and has a link preview, the user double taps to activate the accessibility action, sending the user to the Signal homepage in Safari. The user returns to the Signal app, and taps a second message which contains text along with a link to Signal.org which says "Check out this cool website: https://signal.org". The user double taps to activate the accessibility action and is sent to the Signal homepage in Safari. The user returns to the Signal app. the user taps a third message which contains the same text and link "Check out this cool website: https://signal.org", however this time the message does not contain a generated link preview. The user double taps to activate the accessibility action however nothing happens.] 

[Note that I'm not an experienced VoiceOver user so please so correct me if I'm missing anything]
